packages/api/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Header,
+  Post,
+  Req,
+  HttpException,
+  UnprocessableEntityException,
+  BadRequestException,
+} from "@nestjs/common";
+import {
+  ApiTags,
+  ApiOkResponse,
+  ApiExcludeController,
+  ApiUnprocessableEntityResponse,
+  ApiResponse,
+  ApiBadRequestResponse,
+  ApiNoContentResponse,
+} from "@nestjs/swagger";
+import { swagger } from "../config/featureFlags";
+import { generateNonce, SiweErrorType, SiweMessage } from "siwe";
+import { Request } from "express";
+import { VerifySignatureDto } from "./auth.dto";
+import { ConfigService } from "@nestjs/config";
+import { z } from "zod";
+
+const entityName = "auth";
+
+@ApiTags("Auth BFF")
+@ApiExcludeController(!swagger.bffEnabled)
+@Controller(entityName)
+export class AuthController {
+  constructor(private readonly configService: ConfigService) {}
+
+  @Get("nonce")
+  @Header("Content-Type", "text/plain")
+  @ApiOkResponse({
+    description: "Nonce was returned successfully",
+    schema: { type: "string", example: "2lkRALIfNHY16ZARc" },
+  })
+  public getNonce(@Req() req: Request): string {
+    req.session.nonce = generateNonce();
+    return req.session.nonce;
+  }
+
+  @Post("verify")
+  @Header("Content-Type", "application/json")
+  @ApiOkResponse({
+    description: "Signature was verified successfully",
+    schema: { type: "boolean" },
+  })
+  @ApiBadRequestResponse({
+    description: "Nonce must be requested first",
+    schema: { type: "object", properties: { message: { type: "string" } } },
+  })
+  @ApiBadRequestResponse({
+    description: "Failed to verify signature",
+    schema: { type: "object", properties: { message: { type: "string" } } },
+  })
+  @ApiResponse({
+    status: 440,
+    description: "Signature is expired",
+    schema: { type: "object", properties: { message: { type: "string" } } },
+  })
+  @ApiUnprocessableEntityResponse({
+    description: "Signature is invalid",
+    schema: { type: "object", properties: { message: { type: "string" } } },
+  })
+  public async verifySignature(@Body() body: VerifySignatureDto, @Req() req: Request): Promise<boolean> {
+    if (!req.session.nonce) {
+      throw new BadRequestException({ message: "Nonce must be requested first" });
+    }
+
+    const siweMessage = this.buildSiweMessage(body.message, req);
```

Comment:
For security reasons, should we perhaps generate the message on server side instead of using whatever value is provided in the request?
This could be potentially a security problem, e.g. user signs innocent looking message on a malicious website and then attacker just uses that signature to get the token